site/app/controllers/UserProfileController.php

```diff
@@ -176,8 +176,7 @@ public function changeSecondaryEmail(): JsonResponse {
         if (isset($_POST['secondary_email']) && isset($_POST['secondary_email_notify'])) {
             $secondaryEmail = trim($_POST['secondary_email']);
             $secondaryEmailNotify = trim($_POST['secondary_email_notify']) === "true";
-
-            if ((!$secondaryEmailNotify && $secondaryEmail == "") || $user->validateUserData('user_email_secondary', $secondaryEmail) === true) {
+            if (((!$secondaryEmailNotify && $secondaryEmail == "")) || (($user->validateUserData('user_email_secondary', $secondaryEmail) === true) && !($secondaryEmail == ""))) {
```

Comment:
This could be simplified a bit:
```suggestion
            if ((!$secondaryEmailNotify && $secondaryEmail === "") || ($secondaryEmail !== "" && $user->validateUserData('user_email_secondary', $secondaryEmail) === true)) {
```